src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -2,277 +2,930 @@
+                //     public string M2 => null;
+                Diagnostic(ErrorCode.WRN_NewOrOverrideExpected, "M2").WithArguments("Derived.M2", "Base.M2").WithLocation(10, 19),
+                // (15,19): warning CS0108: 'Derived2.M2' hides inherited member 'Derived.M2'. Use the new keyword if hiding was intended.
+                //     public string M2 => null;
+                Diagnostic(ErrorCode.WRN_NewRequired, "M2").WithArguments("Derived2.M2", "Derived.M2").WithLocation(15, 19),
+                // (20,19): warning CS0108: 'Derived3.M2' hides inherited member 'Derived.M2'. Use the new keyword if hiding was intended.
+                //     public object M2 => null;
+                Diagnostic(ErrorCode.WRN_NewRequired, "M2").WithArguments("Derived3.M2", "Derived.M2").WithLocation(20, 19)
+                );
+            VerifyNoOverride(comp, "Derived.M1");
+            VerifyNoOverride(comp, "Derived.M2");
+            VerifyNoOverride(comp, "Derived2.M1");
+            VerifyNoOverride(comp, "Derived2.M2");
+            VerifyNoOverride(comp, "Derived3.M1");
+            VerifyNoOverride(comp, "Derived3.M2");
+            comp = CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                // (10,19): warning CS0114: 'Derived.M2' hides inherited member 'Base.M2'. To make the current member override that implementation, add the override keyword. Otherwise add the new keyword.
+                //     public string M2 => null;
+                Diagnostic(ErrorCode.WRN_NewOrOverrideExpected, "M2").WithArguments("Derived.M2", "Base.M2").WithLocation(10, 19),
+                // (15,19): warning CS0108: 'Derived2.M2' hides inherited member 'Derived.M2'. Use the new keyword if hiding was intended.
+                //     public string M2 => null;
+                Diagnostic(ErrorCode.WRN_NewRequired, "M2").WithArguments("Derived2.M2", "Derived.M2").WithLocation(15, 19),
+                // (20,19): warning CS0108: 'Derived3.M2' hides inherited member 'Derived.M2'. Use the new keyword if hiding was intended.
+                //     public object M2 => null;
+                Diagnostic(ErrorCode.WRN_NewRequired, "M2").WithArguments("Derived3.M2", "Derived.M2").WithLocation(20, 19)
+                );
+            VerifyNoOverride(comp, "Derived.M1");
+            VerifyNoOverride(comp, "Derived.M2");
+            VerifyNoOverride(comp, "Derived2.M1");
+            VerifyNoOverride(comp, "Derived2.M2");
+            VerifyNoOverride(comp, "Derived3.M1");
+            VerifyNoOverride(comp, "Derived3.M2");
+        }
+
+        [Fact]
+        public void ChainedOverrides_01()
+        {
+            var source = @"
+public class Base
+{
+    public virtual object M1 => null;
+    public virtual object M2 => null;
+    public virtual object M3 => null;
+}
+public class Derived : Base
+{
+    public override string M1 => null;
```

Comment:
>public override string M1 => null; [](start = 4, length = 34)

When there are multiple equal lines, we usually add some distinct comments to them to make it easier to navigate the base-line